gobblin-api/src/main/java/gobblin/configuration/SourceState.java

```diff
@@ -67,6 +69,14 @@ public SourceState(State properties, List<WorkUnitState> previousTaskStates) {
     this.previousTaskStates.addAll(previousTaskStates);
   }
 
+  public List<WatermarkInterval> getPreviousWatermarkIntervals() {
```

Comment:
Yeah, that makes sense. I actually just defined to remove this method altogether because there is already a `SourceState.getPreviousWorkUnitStates()` method, and a `WorkUnitState.getWatermarkInterval`, so this method isn't really necessary.